StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1609SA1610CodeFixProvider.cs

```diff
@@ -0,0 +1,199 @@
+    /// <remarks>
+    /// <para>To fix a violation of this rule, fill-in a description of the value held by the property within the
+    /// &lt;value&gt; tag.</para>
+    /// </remarks>
+    [ExportCodeFixProvider(nameof(SA1609SA1610CodeFixProvider), LanguageNames.CSharp)]
+    [Shared]
+    public class SA1609SA1610CodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+            ImmutableArray.Create(SA1609PropertyDocumentationMustHaveValue.DiagnosticId, SA1610PropertyDocumentationMustHaveValueText.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                if (!FixableDiagnostics.Contains(diagnostic.Id, StringComparer.Ordinal))
+                    continue;
+
+                string description = "Document value from summary";
+                context.RegisterCodeFix(CodeAction.Create(description, cancellationToken => this.GetTransformedDocumentAsync(context.Document, diagnostic, cancellationToken)), diagnostic);
+            }
+
+            return Task.FromResult(true);
+        }
+
+        private async Task<Document> GetTransformedDocumentAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var documentRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            SyntaxNode syntax = documentRoot.FindNode(diagnostic.Location.SourceSpan);
+            if (syntax == null)
+                return document;
+
+            PropertyDeclarationSyntax propertyDeclarationSyntax = syntax.FirstAncestorOrSelf<PropertyDeclarationSyntax>();
+            if (propertyDeclarationSyntax == null)
+                return document;
+
+            DocumentationCommentTriviaSyntax documentationComment = propertyDeclarationSyntax.GetDocumentationCommentTriviaSyntax();
+            if (documentationComment == null)
+                return document;
+
+            XmlElementSyntax summaryElement = documentationComment.Content.GetFirstXmlElement(XmlCommentHelper.SummaryXmlTag) as XmlElementSyntax;
+            if (summaryElement == null)
+                return document;
+
+            SyntaxList<XmlNodeSyntax> summaryContent = summaryElement.Content;
+            if (!this.TryRemoveSummaryPrefix(ref summaryContent, "Gets or sets "))
+            {
+                if (!this.TryRemoveSummaryPrefix(ref summaryContent, "Gets "))
+                {
+                    this.TryRemoveSummaryPrefix(ref summaryContent, "Sets ");
+                }
+            }
```

Comment:
You can't from the return value for sure. That's why I added all the possible prefixes as parameters to the call and they should then match in left to right order I guess.